Source/DafnyCore/Dafny.atg

```diff
@@ -1175,14 +1175,24 @@ ModuleExport<ModuleDefinition parent, out ModuleDecl submodule>
     "provides"
     (
       ( ExportSignature<true, out exsig>        (. exports.Add(exsig); .)
-        {"," ExportSignature<true, out exsig>   (. exports.Add(exsig); .) }
-      )
+        {"," 
+             ( ExportSignature<true, out exsig>  (. exports.Add(exsig); .)
+             | "provides" (. SemErr(la, "no comma is allowed between provides and reveals clauses in an export declaration"); .)
+             | "reveals" (. SemErr(la, "no comma is allowed between provides and reveals clauses in an export declaration"); .)
```

Comment:
Is it possible to look ahead for these keywords and generate an error, but not consume the token?

Reply:
I thought of that, but my attempts to do did not work and got increasingly complicated. I think this will do for now, but better recovery is a future project, and is also part of improving error messages.